autoload/maktaba/syscall.vim

```diff
@@ -271,3 +364,31 @@ function! maktaba#syscall#SetUsableShellRegex(regex) abort
   call maktaba#ensure#IsString(a:regex)
   let s:usable_shell = a:regex
 endfunction
+
+
+""
+" @private
+" Forces the disabling of asynchronous calls, to enable testing.
+function! maktaba#syscall#SetAsyncDisabled(disabled)
+  let s:async_disabled = a:disabled
+endfunction
+
+
+""
+" @public
```

Comment:
Feels more `@private`.

The callback prototype needs to be updated here, too (to `callback(env_dict, result_dict)`).